### ISSUE TITLE
D2ELV dimmer switch control

### ISSUE THREAD

**Issue by maxcredits** [schmittx/home-assistant-leviton-decora-smart-wifi#4]
I'm new to HA. Started adding devices and got to my Leviton dimmer switch **D2ELV**. I installed the integration, and it's picked up all of my Leviton switches, including the dimmer. The dimmer shows all the configuration options, but there is no light control. Am I doing something wrong, or is this something that could be added in the future? 

Thanks for the work you've done.  I'm also new to GitHub, so please let me know if this is the wrong place to put this.

**Comment by schmittx** [comment]
Yeah, I haven't directly tried that specific dimmer so it's not officially supported in this integration. Hence why some entities show up for it, but not everything.

However, I imagine it's pretty similar to the D26HD variant so I could get it officially added with a bit of effort.

If you're willing to share the API responses for it and some screenshots of its config options within the Leviton API, I'll be glad to work on this. I can help you with grabbing the API responses.

**Comment by maxcredits** [comment]
Sure. Let me know what you need and how to get it, and I'd be more than happy to. Thank you!

**Comment by schmittx** [comment]
Great, let's do it.

1) To enable logging of the server (API) responses, do the following.
- Within HA, go to your user profile (bottom left corner of the sidebar menu) and enable "Advanced mode"
- Within HA, go to "Settings > Devices & services > Leviton Decora Smart Wi-Fi > Configure"
- Click through the config windows until you get to "Advanced options", enable to save responses.

2) Restart HA, let the integration fully initialize, then find the "responses" folder that was indicated in the "Advanced options" window. In that folder, look for a file named `residences_******_iotswitches.json` (`******` will be a random number). Within that file, find the block(s) for the D2ELV device(s); please copy/paste here and redact any personal info if desired (there's nothing really sensitive to begin with but it's up to you).

3) In parallel, it's important to understand what the config options are for D2ELV devices within the MyLeviton app. I think the easiest way for me to understand that is if you can share screenshots of the all the possible windows. A lot of the config options are common between the various Leviton devices (ex. "Auto Shutoff", "Status LED Behavior", etc.), so I'm specifically looking for any exclusive parameters that D2ELV has (and other devices don't) or the opposite; any parameters which the D2ELV doesn't support. Worst case, just screen grab every possible window within the settings for the device and I'll wade through it all.

![image](https://github.com/user-attachments/assets/4cd42c35-ef32-4a56-ab35-3a355a0908ad)

![image](https://github.com/user-attachments/assets/09729c80-65e5-412f-8257-75622223d584)

![image](https://github.com/user-attachments/assets/e92f7b8a-c121-4890-85b6-c59b80a439be)

![5F6C2972-CDC6-41FA-B1BA-162811B39EF8_1_101_o](https://github.com/user-attachments/assets/6b3f9c18-f7a3-4775-b336-c8739742a95c)

**Comment by maxcredits** [comment]
OK! Here's my first attempt. Is this the info you are looking for? 

[D2ELV Block.txt](https://github.com/user-attachments/files/18141293/D2ELV.Block.txt)
<img width="430" alt="Dashboard Configuration " src="https://github.com/user-attachments/assets/7255237c-9ae9-4957-ac2a-bc52abd467f0" />
<img width="592" alt="Device Configuration" src="https://github.com/user-attachments/assets/84d639bc-b2b5-41c9-8f43-9ea1008c1dc2" />

**Comment by schmittx** [comment]
Yep, the server response is perfect, exactly what I need.

Regarding the screenshots; I was referring to screenshots of the various options for that device within the My Leviton app on your phone.

**Comment by maxcredits** [comment]
Oh, ok.. there are quite a few. I'll get them together this afternoon.

**Comment by schmittx** [comment]
Yeah, if you've got another non-D2ELV device then you can compare the various config options for both and possibly save some screenshots. I just want to know what's different with D2ELV (options it supports that other devices don't and/or options it doesn't support that other devices do).

**Comment by maxcredits** [comment]
[App Screen Shots.zip](https://github.com/user-attachments/files/18141631/App.Screen.Shots.zip)
I zipped the screenshots. They are in consecutive order, going through the options. Let me know if there is anything else I can do. I greatly appreciate your effort in this and in creating this in the first place.

**Comment by maxcredits** [comment]
[D23LP Dimmer Plug Differences.zip](https://github.com/user-attachments/files/18142396/D23LP.Dimmer.Plug.Differences.zip)

Let me know if that helps. 
Differences between D23LP and D2ELV

**Comment by schmittx** [comment]
Excellent, this is a huge help.

I'm 99% there, but I want to confirm something.

Within the My Leviton app, can you please set the "Bulb Type" to "ELV"; then restart HA and re-share the same code block from before?

Specifically, I want to confirm what the value for `loadType` is when it's set to "ELV". 

`"linkData": "","loadType": 1,
        "localIP": "192.168.1.230",`

**Comment by maxcredits** [comment]
[D2ELV Block (Type ELV).txt](https://github.com/user-attachments/files/18159797/D2ELV.Block.Type.ELV.txt)

"LoadType": 6

**Comment by schmittx** [comment]
I just pushed a new release ([1.6.0](https://github.com/schmittx/home-assistant-leviton-decora-smart-wifi/releases/tag/1.6.0)) with support for D2ELV, let me know if there's any concerns with it.

**Comment by maxcredits** [comment]
I just downloaded and upgraded the integration, and it works **PERFECTLY**. How can I thank you?

**Comment by maxcredits** [comment]
A beginner's question, I think. How do I get HA to refresh (pickup) new devices added in my Leviton App? I added a dimmer plug and cannot see that in HA.

**Comment by schmittx** [comment]
Glad it's working for you!

You have to update the configuration to add new devices; "Settings > Devices & services > Leviton Decora Smart Wi-Fi > Configure". You should see the new devices in the list, just check the box to enable them within HA.

**Comment by maxcredits** [comment]
Yes, I looked for that, and the new device is not showing up. It's an older dimmer plug I had - DW3HL. I have a few of them I was going to try to reuse. Also, I've noticed that the UI update is about 3-7 seconds. I don't know if that's on my end or not. I'm running on a HA green. The UI is not a big deal. Just an FYI.

**Comment by schmittx** [comment]
Found the problem, I made a typo while doing some backend code improvement. I just pushed a new version ([link](https://github.com/schmittx/home-assistant-leviton-decora-smart-wifi/releases/tag/1.6.2)) which fixes this. Please update and confirm the DW3HL shows up.

Regarding the UI update, it could be many things. The integration relies on multiple calls to Eero's cloud-based API every time it needs to update data. I'm not familiar with HA green specs, but a faster host machine might help. Also, try to limit the number of devices, profiles, clients, and activity metrics you select within the integration.

**Comment by maxcredits** [comment]
Thanks, I'll give it a try later this evening. Do you have any recommendations for a host machine? Outside the scope here, but I have no problem moving it if it improves performance and I have a LOT of devices. I have a Home Assistant Yellow that I haven't put together yet. And I have plenty of computers around. Maybe a VM? Love to get your thoughts, but don't feel obligated. I know it's a busy time, and this conversation is not within the scope of this thread. let me know. Thanks
